lib/tapioca/runtime/trackers/required_ancestor.rb

```diff
@@ -10,22 +10,32 @@ module RequiredAncestor
         class << self
           extend T::Sig
 
-          sig { params(requiring: T::Helpers, block: T.proc.returns(Module)).void }
+          sig { params(requiring: T::Helpers, block: T.proc.void).void }
           def register(requiring, block)
             ancestors = @required_ancestors_map[requiring] ||= []
             ancestors << block
           end
 
-          sig { params(mod: Module).returns(T::Array[T.proc.returns(Module)]) }
+          sig { params(mod: Module).returns(T::Array[T.proc.void]) }
           def required_ancestors_blocks_by(mod)
             @required_ancestors_map[mod] || []
           end
 
-          sig { params(mod: Module).returns(T::Array[T.nilable(Module)]) }
+          sig { params(mod: Module).returns(T::Array[T.untyped]) }
           def required_ancestors_by(mod)
             blocks = required_ancestors_blocks_by(mod)
             blocks.map do |block|
-              block.call
+              # Common return values of `block.call` here could be a Module or a Sorbet's runtime value for T.class.
+              # But in reality it could be whatever the block has that can pass Sorbet's static check. Like
+              #
+              # ```
+              # requires_ancestor { T.class_of(Foo); nil }
+              # ```
+              #
+              # So it's designed to be used at runtime and it's accidental that just calling `to_s` on the above common
```

Comment:
Don't you mean it's **not** designed to be used?

Reply:
🤦‍♂️